### PR TITLE
Augment Initialize Endpoint

### PR DESCRIPTION
**Pull request by Untellable**
Augment initialize endpoint to include max nodes, node ranking, and locality parameters to allow better control over node selection. 

I haven't had time to unit test yet, though @Kitwradr has visualized a few of a the results and they look correct.

Note that some connections between nodes may be missing due to mastodon pagination only returning a random subset of relationships, though the result should always form a fully connected graph.

### CODE REVIEW

File: server/src/service/initialize/initialize-service.js [hunk 1]
```diff
@@ -80,85 +82,142 @@ async function accountFollowingService(api, id, db) {
...
+        depth: 0,
+        priority: 1,
+    });
+
+    // Sort nodes by priority using custom max heap
+    const heapComparator = (a, b) => b.priority - a.priority;
+    const nodeHeap = new Heap(heapComparator);
+    nodeHeap.init([accountInfoMap.get(mainId)]);
+
+    let rankFun; // Choose parameter to rank nodes by
```
Comment (alexluo1998) [inline server/src/service/initialize/initialize-service.js:129]: I would probably extract this (and even including the calcPriority) into a seperate function - cleaner and eaiser to test (maybe you are already doing it)
Also might be personal preference but I would use switch case for this

File: server/src/service/initialize/initialize-service.js [hunk 2]
```diff
@@ -80,85 +82,142 @@ async function accountFollowingService(api, id, db) {
...
 
-    // Fetch and process first-level followers and followings
-    const firstLevelFollowers = await accountFollowersService(api, id, db);
-    const firstLevelFollowings = await accountFollowingService(api, id, db);
-    await processAccounts(firstLevelFollowers, true);
-    await processAccounts(firstLevelFollowings, false);
+    // Update node follower info to only contain nodes in accountInfoMap
+    const selectedNodes = new Set(accountInfoMap.keys());
+    for (const nodeInfo of accountInfoMap.values()) {
+        const removeUnusedFollower = (nodeId) =>
```
Comment (alexluo1998) [inline server/src/service/initialize/initialize-service.js:209]: again this is probably just personal preference :) I can also see you way doesn't look like a nested loop, but I will include this anyways...

```
// remove unused followers
nodeInfo['following'].forEach((nodeId) => {
     if (!selectedNodes.has(nodeId)) {
             nodeInfo['following'].delete(nodeId)
     }
});
```

**Review by alexluo1998 — DISMISSED** [review]
some minor style comments

**Review by yunlongwangsd — APPROVED** [review]

File: server/src/service/initialize/initialize-service.js (hunk 2 above)
Comment (Untellable) [reply server/src/service/initialize/initialize-service.js]: good call, updated

File: server/src/service/initialize/initialize-service.js (hunk 1 above)
Comment (Untellable) [reply server/src/service/initialize/initialize-service.js:129]: Yeah I was planning on splitting up the function for unit tests, just didn't get around to it. It's a bit hard bc. there's a lot of variables being using across the function.

**Review by yunlongwangsd — APPROVED** [review]

**Review by keran-w — APPROVED** [review]

**Pull request merged**